### PR TITLE
get rid of species sources by default

### LINKED ISSUE #350: remove source terms for species
In general, we don't need to have the ability to have source terms for the species.  We can make it a compile-time option, off by default, and this will save a lot of memory, and work (tracing).  This can be part of #256

### PR DESCRIPTION
**Pull request by zingale**
This change eliminates species sources by requiring us to set the parameter `PRIM_SPECIES_HAVE_SOURCES`.  To support this the `set_variables.py` script was updated to now allow the "adds-to" field to have an optional if-def requirement.  

This change also fixes 2 bugs.  In `srctoprim`, we were converting the velocity sources twice.  And in the radiation tracing, we were not adding species sources.

This closes #350 

Note: this should be merged after #517

### CODE REVIEW

**Comment by zingale** [comment]
This is the test suite output:  http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2019-02-19-005/index.html

File: Source/hydro/advection_util_nd.F90 [hunk 1]
```diff
@@ -692,6 +692,13 @@ subroutine ca_srctoprim(lo, hi, &
        n = upass_map(ipassive)
        iq = qpass_map(ipassive)
 
+       ! we already accounted for velocities above
+       if (n == QU .or. n == QV .or. n == QW) cycle
```
Comment (maxpkatz) [inline Source/hydro/advection_util_nd.F90]: I think you should be testing on iq here, not n?

Comment (zingale) [reply Source/hydro/advection_util_nd.F90]: yes.  Thankfully the velocity indices are the same.  Wonder if this is wrong elsewhere...

Comment (zingale) [reply Source/hydro/advection_util_nd.F90]: I checked the other places and it looked okay elsewhere.

**Pull request merged**